docs/configuration/integrations/integrations-next/eventhandler-config.md

```diff
@@ -0,0 +1,300 @@
+
+To use the cache feature and maintain state in a Kubernetes environment, a [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) must be used. Sample manifests are provided at the bottom of this doc. Please adjust these according to your deployment preferences. You can also use a Deployment, however the presence of the cache file will not be guaranteed and the integration may ship duplicate entries in the event of a restart. Loki does not yet support entry deduplication for the A->B->A case, so further deduplication can only take place at the Grafana / front-end layer (Grafana Explore does provide some deduplication features for Loki datasources).
+
+This integration uses Agent's embedded Loki-comptaible `logs` subsystem to ship entries, and a logs client and sink must be configured to use the integration. Please see the sample Agent config below for an example configuration. [Pipelines](https://grafana.com/docs/loki/latest/clients/promtail/pipelines/) and relabel configuration are not yet supported, but these features will be added soon. You should use the `job=eventhandler cluster=...` labels to query your events (you can then use LogQL on top of the result set).
+
+If not running the integration in-cluster, the integration will use `kubeconfig_path` to search for a valid Kubeconfig file, defaulting to a kubeconfig in the user's home directory. If running in-cluster, the appropriate `ServiceAccount` and Roles must be defined. Sample manifests are provided below.
+
+Configuration reference:
+
+```yaml
+  ## Eventhandler hands watched events off to promtail using a promtail
+  ## client channel. This parameter configures how long to wait (in seconds) on the channel
+  ## before abandoning and moving on.
+  [send_timeout: <int> | default = 60]
+
+  ## Configures a cluster= label to add to log lines
+  [cluster_name: <string> | default = "cloud"]
+
+  ## Configures the path to a kubeconfig file. If not set, will fall back to using 
+  ## an in-cluster config. If this fails, will fall back to checking the user's home
+  ## directory for a kubeconfig.
+  [kubeconfig_path: <string>]
+
+  ## Path to a cache file that will store the last timestamp for a shipped event and events
+  ## shipped for that timestamp. Used to prevent double-shipping on integration restart.
+  [cache_path: <string> | default = "./.eventcache/eventhandler.cache"]
+
+  ## Name of logs subsystem instance to hand log entries off to.
+  [logs_instance: <string> | default = "default"]
+
+  ## K8s informer resync interval (seconds). You should use defaults here unless you are 
+  ## familiar with K8s informers.
+  [informer_resync: <int> | default = 120]
+
+  ## The integration will flush the last event shipped out to disk every flush_interval seconds.
+  [flush_interval: <int> | default = 10]
+
+  ## If you would like to limit events to a given namespace, use this parameter.
+  [namespace: <string>]
```

Comment:
Just curious, are the options for namespace filtering currently limited to all or 1?

Reply:
Yes, due to https://github.com/kubernetes/kubernetes/issues/74415 (and others).. if there's sufficient demand for list input I can implement it